pkg/keccak/Makefile

```diff
@@ -0,0 +1,24 @@
+PKG_NAME    := keccak
+PKG_URL     := https://github.com/gvanas/KeccakCodePackage
+PKG_VERSION := bc592d1780a6b3204c26c6f872552ace8fc6fd65
+
+ifeq ($(CPU),native)
+  CPU_ARCHIVE_NAME := Native
+else
+ifeq ($(CPU_ARCH),$(filter $(CPU_ARCH), cortex-m0 cortex-m0plus))
+  CPU_ARCHIVE_NAME := ARMv6M
+else
+ifeq ($(CPU_ARCH),$(filter $(CPU_ARCH), cortex-m3 cortex-m4 cortex-m4f cortex-m7 cortex-m7f))
+  CPU_ARCHIVE_NAME := ARMv7M
+endif
+endif
+endif
+
+.PHONY: all
+
+all:
+	@cp -r $(RIOTBASE)/pkg/keccak/contrib/Modes $(PKG_BUILDDIR)
+	$(MAKE) KeccakHash$(CPU_ARCHIVE_NAME).a -C $(PKG_BUILDDIR)
+	@cp $(PKG_BUILDDIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a $(BINDIR)/keccak.a
+
+include $(RIOTBASE)/pkg/pkg.mk
```

Comment:
this line needs to move up, to right below the PKG_ variables.

Reply:
@aabadie can you advise how to change this so it supports out-of-tree builds?

Reply:
The way it's done won't support out-of-tree builds since it uses the build system provided by Keccak. But I think that could maybe be avoided and if that's really the case, that will simplify a bit the package (no build system patch needed).

Reply:
I made a list of comment with suggestions that works for me when I build the package locally. There's room for improvements: the Makefile, as it is written now, will execute all commands in the `all` each time the an application using this package is built. This can be avoid by making all to depend on `$(BINDIR)/keccak.a`, which itself would depend on `$(PKG_BUILD_DIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a` and run the `cp` command.
Then, finally, the `$(PKG_BUILD_DIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a` target, will can the build command (we could also avoid the copy of `$(RIOTBASE)/pkg/keccak/contrib/Modes`.

I also noticed that a test script was missing. It should be pretty straight-forward to add, with the following content:
```py
#!/usr/bin/env python3

# Copyright (C) 2016 Author Name <author.name@mail.org>
#
# This file is subject to the terms and conditions of the GNU Lesser
# General Public License v2.1. See the file LICENSE in the top level
# directory for more details.

import sys
from testrunner import run_check_unittests


if __name__ == "__main__":
    sys.exit(run_check_unittests())
```

Reply:
I do see a `run-01.py` in `tests/pkg_keccak/tests`, but I can update it to look like your proposal.